src/charm.py

```diff
@@ -197,7 +197,12 @@ def _handle_ingress(self, interfaces):
             )
 
     def _check_kf_profiles(self, interfaces):
-        if not ((kf_profiles := interfaces["kubeflow-profiles"]) and kf_profiles.get_data()):
+        kf_profiles = interfaces["kubeflow-profiles"]
+
+        if not kf_profiles:
+            raise CheckFailed("Relation to kubeflow-profiles required", BlockedStatus)
```

Comment:
I got a suggestion from @beliaev-maksim: if blocking a unit means a human operator needs to do something on it, maybe the message could be an imperative statement, rather than just informative. Something like "add relation".

Reply:
```suggestion
            raise CheckFailed("Add required relation to kubeflow-profiles", BlockedStatus)
```